Rubberduck.Parsing/VBA/Parsing/AttributeListener.cs

```diff
@@ -52,13 +56,23 @@ public override void ExitSubStmt(VBAParser.SubStmtContext context)
                 _attributes.Add(_currentScope, _currentScopeAttributes);
                 context.AddAttributes(_currentScopeAttributes);
             }
+
+            ResetScope();
+        }
+
+        private void ResetScope()
```

Comment:
I'm not clear why we need to reset the scope? Or is it more like update the scope?

Reply:
This ensures that we do not pick up procedure attributes outside of their contexts be resetting the current scope and current scope attributes to their inital values.

Reply:
I wonder if it'll help with understanding the intention if we used `PushScope` and PopScope` instead of an assignment statement in the entering rule and a call to `ResetScope` in the exiting rule?